### PR TITLE
Fixed issue #17790: KCFinder Image Browse Server not working due to i…

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue #17790: KCFinder Image Browse Server not working due to incorrect cookieDomain
Dev: this need to set it in config.

### CODE REVIEW

**Review by gabrieljenik — APPROVED** [review]

File: application/helpers/admin/htmleditor_helper.php [hunk 1]
```diff
@@ -27,8 +27,11 @@ function initKcfinder()
         'flash' => $sAllowedExtensions,
         'images' => $sAllowedExtensions
     );
-    if (Yii::app()->getRequest()->enableCsrfValidation && !empty(Yii::app()->getRequest()->csrfCookie->domain)) {
-        $_SESSION['KCFINDER']['cookieDomain'] = Yii::app()->getRequest()->csrfCookie->domain;
+    if (!empty(App()->getSession()->cookieParams['domain'])) {
+        $_SESSION['KCFINDER']['cookieDomain'] = App()->getSession()->cookieParams['domain'];
+    }
+    if (App()->getRequest()->enableCsrfValidation && !empty(App()->getRequest()->csrfCookie['domain'])) {
```
Comment (gabrieljenik) [inline application/helpers/admin/htmleditor_helper.php:33]: Should we use `App()` or `Yii::app()` ?

**Review by gabrieljenik — APPROVED** [review]

File: application/helpers/admin/htmleditor_helper.php (hunk 1 above)
Comment (Shnoulle) [reply application/helpers/admin/htmleditor_helper.php:33]: Never sure …

**Comment by operatorone** [comment]
Hi there, I updated to the Version 3.28.6 and get constantly the Error CSRF upload error if I try to upload PDF or image diles. I plan to update to Version 5.X, but important surveys are running currently, so I can't update. Is there any fix to get around this error for the time being?

Screenshot https://cln.sh/MVKNQqw98IRy1OyTDCqy

**Comment by gabrieljenik** [comment]
Hi, I am bit confused.
Sorry, the error you are reporting, is it the same as # 17790? If so this patch is for master, not for  3.28.6
If not, please, report the new error on bugs.limesurvey.org

Thanks

**Comment by operatorone** [comment]
Yes, I guess it is the same issue. I can't just upload anything to lime survey, because of the cookie error.

**Comment by Shnoulle** [comment]
It's in 3.X https://github.com/LimeSurvey/LimeSurvey/commit/8fa2388ea48b346e4158717415678b329cf08a9c

BUT : you need to set domain in your config file : https://manual.limesurvey.org/Optional_settings/en#Other_sessions_update

If it not found directly.

**Comment by operatorone** [comment]
I added the following lines to the file /application/config/config.php

`     // Set the domain for cookie
		 'session' => array (
			'cookieParams' => array(
							'domain' => '.domainname.com',
					),
			),
`

Limesurvey runs an o subdomain called survey.domainname.com
Unfortunately the CSRF error still appears.

**Comment by operatorone** [comment]
Can you reopen the ticket, so I can comment.

**Comment by Shnoulle** [comment]
Ask on [forum](https://forums.limesurvey.org/), and look at your the bug report : https://bugs.limesurvey.org/view.php?id=17790

You line is false here, this exact issue is fixed . Maybe there are something else.

**Pull request merged**